lib/db.js

```diff
@@ -418,12 +420,25 @@ module.exports = function (
 
   DB.prototype.updateSessionTokenInBackground = function (token, userAgentString) {
     log.trace({ op: 'DB.updateSessionTokenInBackground', uid: token && token.uid })
-    return this.pool.post(
-      '/sessionToken/' + token.id + '/update',
-      userAgent.call({
-        lastAccessTime: Date.now()
-      }, userAgentString)
-    )
+
+    var freshData = userAgent.call({
+      lastAccessTime: Date.now()
+    }, userAgentString)
+
+    if (isSessionTokenFresh(token, freshData)) {
+      return P.resolve()
+    }
+
+    return this.pool.post('/sessionToken/' + token.id + '/update', freshData)
+  }
+
+  function isSessionTokenFresh (token, freshData) {
+    return token.uaBrowser === freshData.uaBrowser &&
+      token.uaBrowserVersion === freshData.uaBrowserVersion &&
+      token.uaOS === freshData.uaOS &&
+      token.uaOSVersion === freshData.uaOSVersion &&
+      token.uaDeviceType === freshData.uaDeviceType &&
+      token.lastAccessTime + ONE_HOUR > freshData.lastAccessTime
   }
```

Comment:
This `isSessionTokenFresh` logic feels out of place here, I suggest refactoring it into a method on the `SessionToken` class.  It could be just an `isFresh` method like you've got here, or perhaps a `token.update()` method that further encapsulates the details of pulling fields from the user-agent header, and returns a boolean to indicate whether the data was changed.